profile/encode.go

```diff
@@ -304,6 +320,12 @@ func (p *Profile) postDecode() error {
 		}
 		if len(numLabels) > 0 {
 			s.NumLabel = numLabels
+			for key, units := range numUnits {
+				if lenUnits, lenLabels := len(units), len(numLabels[key]); lenUnits > 0 {
```

Comment:
You don't need the temporary variables because you only use the value once. It should just be
```
if len(units) > 0 {
  numUnits[key] = padStringArray(units, len(numLabels[key]))
}
```